packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts

```diff
@@ -0,0 +1,952 @@
+
+  if (!hiddenInput) {
+    throw new Error(`Switch with name "${name}" not found`);
+  }
+
+  return hiddenInput;
+}
+
+function setInputValueByLabel(
+  labelText: string,
+  value: string,
+  root: Document | HTMLElement = document,
+): HTMLInputElement {
+  const input = getInputByLabel(labelText, root);
+  if (!input) {
+    throw new Error(`Input with label "${labelText}" not found`);
+  }
+  input.value = '';
+
+  value.split('').forEach((char) => {
+    input.value += char;
+    input.dispatchEvent(new Event('input', { bubbles: true }));
+    input.dispatchEvent(new KeyboardEvent('keydown', { bubbles: true, key: char }));
+    input.dispatchEvent(new KeyboardEvent('keypress', { bubbles: true, key: char }));
+    input.dispatchEvent(new KeyboardEvent('keyup', { bubbles: true, key: char }));
+  });
+
+  input.dispatchEvent(new Event('change', { bubbles: true }));
+  input.dispatchEvent(new Event('input', { bubbles: true }));
+
+  return input;
+}
+
+const getPopup = (): NodeListOf<Element> => document.querySelectorAll('.dx-overlay-wrapper.dx-scheduler-appointment-popup');
+
+function openPopupByDblClick(POM: SchedulerModel, text?: string): AppointmentModel {
+  const appointment = POM.getAppointment(text) as AppointmentModel;
+
+  if (!appointment?.element) {
+    throw new Error(`Appointment "${text}" not found`);
+  }
+
+  appointment.element.click();
+  appointment.element.click();
+
+  return appointment;
+}
+
+function selectRadio(container: HTMLElement, value: string): Element | null {
+  const group = container.querySelector('[role="radiogroup"]');
+  if (!group) throw new Error('Radiogroup not found');
+
+  const radios = Array.from(group.querySelectorAll('[role="radio"]'));
+
+  const target = radios.find((radio) => {
+    const label = radio.getAttribute('aria-label')?.trim();
+    const text = radio.textContent?.trim();
+    return label === value || text === value;
+  });
+
+  if (!target) throw new Error(`Radio with value "${value}" not found`);
+
+  radios.forEach((r) => {
+    r.setAttribute('aria-checked', 'false');
+    r.classList.remove('dx-item-selected', 'dx-radiobutton-checked');
+  });
+
+  target.setAttribute('aria-checked', 'true');
+  target.classList.add('dx-item-selected', 'dx-radiobutton-checked');
+
+  target.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+
+  return target;
+}
+
+function getSelectedRadio(container: HTMLElement): HTMLElement | null {
+  return container.querySelector('[role="radio"][aria-checked="true"]');
+}
+
+function getSelectedRadioValue(container: HTMLElement): string | null {
+  const selected = getSelectedRadio(container);
+  return selected?.getAttribute('aria-label') ?? selected?.textContent?.trim() ?? null;
+}
+
+describe('Appointment popup form', () => {
+  beforeEach(() => {
+    fx.off = true;
+    setupSchedulerTestEnvironment();
+  });
+
+  afterEach(() => {
+    fx.off = false;
+    document.body.innerHTML = '';
+    jest.useRealTimers();
```

Comment:
should we create function `cleanSchedulerTestEnvironment` and move this code there?